### ISSUE TITLE
store git hash of the current build

### ISSUE THREAD

**Issue by olamy** [eclipse/jetty.project#1956]
To be able to know which snapshot is used (during test or load test).
Ideally we could store git hash and build timestamp.
```
buildNumber=${buildNumber}
timestamp=${timestamp}
version=${project.version}
```
Can be achieved using the mojo 
```
        <groupId>org.codehaus.mojo</groupId>
        <artifactId>buildnumber-maven-plugin</artifactId>
```
* Adding it in jetty-server module. (so we can display those at start?)
* Adding a sys prop "jetty.git.hash" (similar as we have today with "jetty.version")

WDYT?

**Comment by joakime** [comment]
So this would be a properties file in the jetty-server.jar then?
Something the startup of jetty could use and report in logging?
Sounds like a good idea.
Would definitely help with debugging and reports from SNAPSHOT users too.
Can we show the timestamped SNAPSHOT version?
Also, is buildNumber == git commit hash?

**Comment by joakime** [comment]
Hmm, I wonder if the scm/url could be included as well. (to know if the one we are troubleshooting is from an odd source, like a linux packaging effort)

**Comment by olamy** [comment]
yup easy to add.
Will be part of jetty-server.jar and yes can be log along with the current jetty version printed.
scm url is easy to include.
which branches 9.3.x as well?

**Comment by olamy** [comment]
```
2017-11-10 13:17:04.461:INFO:oejs.Server:main: jetty-9.4.8-SNAPSHOT
2017-11-10 13:17:04.461:INFO:oejs.Server:main: build timestamp: 2017-11-10T13:12:37+11:00, git hash: 7ba4428e9fbb6107a9aa69ae026578be78ae7aa1
``` 
build.properties  contains:
```
buildNumber=7ba4428e9fbb6107a9aa69ae026578be78ae7aa1
timestamp=1510279957275
version=9.4.8-SNAPSHOT
scmUrl=scm:git:https://github.com/eclipse/jetty.project.git/jetty-server
```

**Comment by olamy** [comment]
implemented in 9.3/9.4/master
